### PR TITLE
[MM-61100] Fix errcheck issues in server/channels/app/busy_test.go

### LINKED ISSUE #28780: Fix errcheck issues in server/channels/app/busy_test.go
At Mattermost, we are committed to maintaining a high standard of code quality, reliability, and security. To further enhance this, we are launching a community campaign to integrate the [`errcheck`](https://github.com/kisielk/errcheck) linter across the Mattermost codebase.

This Help Wanted ticket is about fixing the linter issues reported by `errcheck` in `channels/app/busy_test.go`.

### What is `errcheck`?
`errcheck` is a tool that automatically identifies places in Go code where errors are returned but not checked. In Go, error handling is a critical part of writing robust applications, and ignoring returned errors can lead to bugs, crashes, or unintended behaviour. By integrating `errcheck` into the codebase, we can ensure that error handling is consistently and properly managed across the project.

### Tasks
- Remove the exception for `channels/app/busy_test.go` from the exceptions list for `errcheck` in `server/.golangci.yml`.
- Run `make golangci-lint` to get a list of all linter errors.
- Fix the errors by properly handling them.
- Run `make golangci-lint` again to ensure all errors are fixed.

#### Okay, but what does "properly handling" mean?
In general, errors should be returned to the caller. Non critical errors should be loggged, and function execution should continue. There are some exceptions to this rule, such as when the error is expected and should be ignored or when the error is handled in a different way. In such cases, a comment should be added to explain why the error is ignored.

In tests, errors should be checked using `require.NoError(t, err)` for errors of type `error` or `require.Nil(t, err)` for errors of type `**model.AppError`.
If an error is expected, it should be checked using `require.Error(t, err)` for errors of type `error` or `require.NotNil(t, err)` for errors of type `**model.AppError`.

Use `err` as the variable name for errors and `appErr` for `*model.AppError`. Avoid using variable names like `err1`, `err2`, etc. See [the Mattermost style guide](https://developers.mattermost.com/contribute/more-info/server/style-guide/#error-variable-names) for more information.

https://github.com/mattermost/mattermost/pull/28364 and https://github.com/mattermost/mattermost/pull/28376 can be used as examples of how to fix `errcheck` issues in the `api4` package.

-------------

If you're interested, please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on the Mattermost Community server, where you can discuss questions with community members and the Mattermost core team.
For technical advice or questions, please [join our "Developers Server" community channel](https://community.mattermost.com/core/channels/developers-server).
You can also reach out to @hanzei either on this ticket or on [the Mattermost Community server](https://community.mattermost.com).

For new contributors, please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

If this ticket is already taken, please check out our other [Help Wanted tickets](https://github.com/mattermost/mattermost/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+errcheck+label%3A%22Up+For+Grabs%22) of this campaign.

Thank you for your contributions to Mattermost!

----------------

JIRA: MM-61100

### PR DESCRIPTION
**Pull request by RS-labhub**
#### Summary
Fixed the errcheck issues

#### Ticket Link
fixes #28780 
Jira [MM-61100](https://mattermost.atlassian.net/browse/MM-61100)

#### Release Note
```release-note
NONE
```

[MM-61100]: https://mattermost.atlassian.net/browse/MM-61100?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/channels/app/busy_test.go [hunk 1]
```diff
@@ -123,7 +123,9 @@ type ClusterMock struct {
 
 func (c *ClusterMock) SendClusterMessage(msg *model.ClusterMessage) {
 	var sbs model.ServerBusyState
-	json.Unmarshal(msg.Data, &sbs)
+	if err := json.Unmarshal(msg.Data, &sbs); err != nil {
+		return
+	}
```
Comment (hanzei) [inline server/channels/app/busy_test.go:128]: Simply returning could obfuscate why a test is failing, The test should either `panic` or we add a `*testing.T` to `ClusterMock` and simply check using `require.NoError`. I prefer the later.

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: server/channels/app/busy_test.go (hunk 1 above)
Comment (RS-labhub) [reply server/channels/app/busy_test.go]: Done!

**Review by RS-labhub — COMMENTED** [review]
Done!

File: server/channels/app/busy_test.go [hunk 2]
```diff
@@ -119,11 +119,14 @@ func compareBusyState(t *testing.T, busy1 *Busy, busy2 *Busy) bool {
 // ClusterMock simulates the busy state of a cluster.
 type ClusterMock struct {
 	Busy *Busy
+	t    *testing.T
 }
 
 func (c *ClusterMock) SendClusterMessage(msg *model.ClusterMessage) {
 	var sbs model.ServerBusyState
-	json.Unmarshal(msg.Data, &sbs)
+	if err := json.Unmarshal(msg.Data, &sbs); err != nil {
```
Comment (hanzei) [inline server/channels/app/busy_test.go:127]: No need for the `if` clause :wink:

File: server/channels/app/busy_test.go [hunk 3]
```diff
@@ -119,11 +119,14 @@ func compareBusyState(t *testing.T, busy1 *Busy, busy2 *Busy) bool {
 // ClusterMock simulates the busy state of a cluster.
 type ClusterMock struct {
 	Busy *Busy
+	t    *testing.T
```
Comment (hanzei) [inline server/channels/app/busy_test.go:122]: Where does this get set?

File: server/channels/app/busy_test.go (hunk 1 above)
Comment (RS-labhub) [reply server/channels/app/busy_test.go]:
```go
func TestSendClusterMessage(t *testing.T) {
    busy := &Busy{}
    clusterMock := &ClusterMock{
        Busy: busy,
        t:    t,
    }

    msg := &model.ClusterMessage{
        Data: []byte(`{"some": "data"}`),
    }

    clusterMock.SendClusterMessage(msg)
}
```

is this good?

File: server/channels/app/busy_test.go (hunk 3 above)
Comment (RS-labhub) [reply server/channels/app/busy_test.go:122]:
```go
func TestSendClusterMessage(t *testing.T) {
    busy := &Busy{}
    clusterMock := &ClusterMock{
        Busy: busy,
        t:    t,
    }

    msg := &model.ClusterMessage{
        Data: []byte(`{"some": "data"}`),
    }

    clusterMock.SendClusterMessage(msg)
}
```

May I add something like this?

Comment (hanzei) [reply server/channels/app/busy_test.go:122]: There are test like `TestBusySet` which use a `ClusterMock`. You need to add the `testing.T` there.

https://github.com/mattermost/mattermost/blob/63c97f5a6d8f55eb7583ad975ce12a20d6478273/server/channels/app/busy_test.go#L18-L19

**Comment by RS-labhub** [comment]
Could you review it now?

**Review by hanzei — CHANGES_REQUESTED** [review]
There are a couple of other places like https://github.com/mattermost/mattermost/blob/9d2b6a15cf1eeb867e49598485bb20596d02dbf7/server/channels/app/busy_test.go#L57 where `ClusterMock` is initialized. You need to update these as well.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**